code/parachain/frame/cosmwasm/src/runtimes/wasmi.rs

```diff
@@ -473,25 +426,33 @@ impl<'a, T: Config> VMBase for CosmwasmVM<'a, T> {
 		}
 	}
 
-	fn db_scan(
+	fn addr_validate(&mut self, input: &str) -> Result<Result<(), Self::Error>, Self::Error> {
```

Comment:
why Result of Result?

Reply:
This change is a reorder by rust analyzer and unrelated to the pr. Other than that, it's a double result because the inner one does not halt execution and is returned to the contract. The outer one would halt exec